src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java

```diff
@@ -183,6 +197,60 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 		}
 	}
 
+    private void validateOutputPath(String outputPath, String expectedMimeType) {
+        try {
+            Path path = Paths.get(outputPath);
+
+            if (Files.isSymbolicLink(path)) {
+                throw new SecurityException("The input ZIP file must not be a symlink for security reasons");
+            }
+
+            if (Files.exists(path, LinkOption.NOFOLLOW_LINKS)) {
+                BasicFileAttributes attr = Files.readAttributes(
+                    path, BasicFileAttributes.class, LinkOption.NOFOLLOW_LINKS);
+
+                if (!attr.isRegularFile()) {
+                    throw new SecurityException(
+                        "The specified output file " + outputFilePath + " exists, but is not a regular file. Cannot be overwritten.");
+                } else if (!tika.detect(path).equals(expectedMimeType)) {
+                    throw new SecurityException(
+                        "The specified output file " + outputFilePath + " exists, but is not a " + expectedMimeType
+                                + " file as expected, detected type was " + tika.detect(path) + ". Cannot be overwritten");
+                }
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Could not validate input ZIP file path", e);
+        }
+    }
+
+    private void validateInputZipFile(String inputZipFilePath) {
+        try {
+            Path path = Paths.get(inputZipFilePath);
+
+            if (Files.isSymbolicLink(path)) {
+                throw new SecurityException("The input ZIP file must not be a symlink for security reasons");
+            } else if (!Files.exists(path, LinkOption.NOFOLLOW_LINKS)) {
+                throw new IllegalArgumentException("Specified input ZIP file " + inputZipFilePath + " does not exist");
+            }
+
+            BasicFileAttributes attr = Files.readAttributes(
+                path, BasicFileAttributes.class, LinkOption.NOFOLLOW_LINKS);
+
+            if (!attr.isRegularFile()) {
+                throw new SecurityException("The input ZIP file must be a regular file");
+            } else if (attr.size() > MAX_ZIP_FILE_SIZE) {
+                throw new SecurityException("For security, input ZIP files over 600M are not accepted. "
+                        + "This should permit RF2 ALL or SNAPSHOT bundles requiring the required files");
+            } else if (!tika.detect(path).equals("application/zip")) {
+                throw new SecurityException(
+                    "TThe input ZIP file " + outputFilePath + " is not a zip file as expected, detected type was "
```

Comment:
Should this reference `inputZipFilePath` instead of `outputFilePath`

Reply:
thanks again, I'll fix this